application/src/main/java/org/restcomm/sbc/managers/LocationManager.java

```diff
@@ -61,75 +63,76 @@ public static LocationManager getLocationManager() {
 	}
 	
 	
-	public Location register(String user, String host, int port, String userAgent, String transport, int ttl) {
-		Location location=new Location();
-		location.setHost(host);
-		location.setPort(port);
+	public void register(Location location, String userAgent, int ttl) {
+	
 		location.setUserAgent(userAgent);
-		location.setTransport(transport);
-		location.setUser(user);
-		location.setDmzExpirationTimeInSeconds(ttl);
-		location.setMzExpirationTimeInSeconds(ttl);
-		registers.put(user, location, ttl, TimeUnit.SECONDS);
+		location.setExpirationTimeInSeconds(ttl);
+		registers.put(location.getUser(), location, ttl, TimeUnit.SECONDS);
+		if(LOG.isDebugEnabled()) {
+			LOG.debug("registers.put "+location.getUser());
+		}
 		
-		return location;
-			
 	}
 	
+	public Location unregister(String user) {
+		return registers.remove(user);
+	}
+	
+	
 	public Location getLocation(String user) {
-		return (Location) registers.get(user);
+		return registers.get(user);
+	}
+	
+	public Location getLocation(String user, String host) {

```

Comment:
we should implements equals and hashcode and use registers.get(key) instead of iterating and comparing all values of the map. If the register database grow big it will fail because of performance.